### PR TITLE
feat: support split write request to batches for small wal logs

### PR DESCRIPTION
**Pull request by ShiKaiWi**
## Which issue does this PR close?

Closes #

## Rationale for this change
 Currently, one large write request may lead to one large WAL log, which is not friendly for some wal implementations.

## What changes are included in this PR?
- Support split write table request to multiple batches.

## Are there any user-facing changes?
None.

## How does this change test
New unit tests and existing integraion tests.

### CODE REVIEW

File: analytic_engine/src/instance/write.rs [hunk 1]
```diff
@@ -150,6 +150,107 @@ impl EncodeContext {
...
+    /// `max_bytes_per_batch`.
+    max_bytes_per_batch: usize,
+}
+
+enum SplitResult<'a> {
+    Splitted {
+        encoded_batches: Vec<Vec<ByteVec>>,
+        row_group_batches: Vec<RowGroupSlicer<'a>>,
+    },
+    Integrate {
```
Comment (jiacai2050) [inline analytic_engine/src/instance/write.rs:168]: It seems this item could be implemented using Splitted with len is 1?
```
struct SplitResult {
  encoded_batches: Vec<Vec<ByteVec>>,
  row_group_batches: Vec<RowGroupSlicer<'a>>,
}
```

Comment (ShiKaiWi) [reply analytic_engine/src/instance/write.rs:168]: This will lead to extra cost on allocating a vector if no split happens.

Comment (jiacai2050) [reply analytic_engine/src/instance/write.rs:168]: That's one small point, I'm not strong with this, just want to remove some duplicated code below.

**Review by jiacai2050 — APPROVED** [review]
LGTM

**Pull request merged**